pkg/utils/tekton/controller.go

```diff
@@ -111,11 +113,36 @@ func (k KubeController) WatchTaskPod(tr string, taskTimeout int) error {
 	return k.Commonctrl.WaitForPod(k.Commonctrl.IsPodSuccessful(pod.Name, k.Namespace), time.Duration(taskTimeout)*time.Second)
 }
 
+func (k KubeController) GetTaskRunResult(tr *v1beta1.TaskRun, result string) (string, error) {
+	for _, trResult := range tr.Status.TaskRunResults {
+		if trResult.Name == result {
+			return trResult.Value, nil
+		}
+	}
+	return "", fmt.Errorf(
+		"result %q not found in TaskRun %s/%s", result, tr.ObjectMeta.Namespace, tr.ObjectMeta.Name)
+}
+
 func (k KubeController) RunVerifyTask(taskName, image string, taskTimeout int) (*v1beta1.TaskRun, error) {
 	tr := verifyTaskRun(image, taskName)
 	return k.createAndWait(tr, taskTimeout)
 }
 
+type VerifyECTaskParams struct {
```

Comment:
I like the struct. I should have done this for the other tasks. Just a question though, should the code be changed to . The struct and the `GetVerifyECTask` would be moved to tasks.go
```
func (k KubeController) RunVerifyECTask(tr *v1beta1.TaskRun, taskTimeout int) (*v1beta1.TaskRun, error) {
	return k.createAndWait(tr, taskTimeout)
}

func (v VerifyECTaskParams) GetVerifyECTask() (*v1beta1.TaskRun) {
	return verifyEnterpriseContractTaskRun(v)
}
```

Reply:
I tried something like this in a new commit. Let me know what you think!